### ISSUE TITLE
Remove the unused plugins from the final file

### ISSUE THREAD

**Issue by lionel-bijaoui** [xdan/jodit#557]
Hello,

Jodit is very good, but it is rather large (around 600ko).

Is there a way to load a special build of Jodit without its plugins, and load them independently (like custom plugins) ?
Alternatively, maybe there is a way to tap into webpack tree-shaking capabilities to remove unused code ?

The goal here would be to reduce the size of the final file.

Thank you for your time !

**Comment by xdan** [comment]
@lionel-bijaoui  Yes, I thought about it. Jodit supports loading external plugins, I need to fix the webpack config so that each plugin is bundled into its own folder.
But now I don't have time for that. If anyone can help with this, I would be grateful.
Plus, it is probably necessary that all language files are also kept separately. Jodit also supports downloading them.

Alternatively, in the first solution now it would be possible to make a separate file in which to write a list of files that webpack should not include in the assembly. I think the webpack has a similar feature.

**Comment by lukewarlow** [comment]
Not directly related but now #621 is merged there's a build output with just the English language included, which will help with the bundle size issue.

Though of course further work to have an even more minimal build output would be beneficial.

**Comment by xdan** [comment]
This has been possible for a while now, so I'm closing it.

The ESM build pulls in only a basic set of plugins and the English language, and you add what you actually need on top:

```js
import { Jodit } from 'jodit/esm/index.js';
import 'jodit/esm/plugins/fullsize/fullsize.js';
import 'jodit/esm/plugins/add-new-line/add-new-line.js';
// or everything at once: import 'jodit/esm/plugins/all.js';
```

Each plugin lives in its own file, so your bundler keeps out whatever you don't import. Languages work the same way (`jodit/esm/langs/de.js`).

If you'd rather not deal with a bundler, there's also the builder — tick the plugins you want and download a ready-made file: https://xdsoft.net/jodit/builder/?editorType=pro
